tests/core/test_int53.c

```diff
@@ -0,0 +1,412 @@
+#endif
+}
+
+void writeI53ToI64Signaling_double(int64_t *heapAddress, double num)
+{
+#ifdef ANSWERS
+	*heapAddress = (int64_t)num;
+#else
+	EM_ASM(try {
+		writeI53ToI64Signaling($0, $1)
+	} catch(e) {
+		HEAPU32[$0>>2] = 0x01020304;
+		HEAPU32[$0+4>>2] = 0x01020304;
+	}, heapAddress, num);
+#endif
+}
+
+void writeI53ToU64Clamped_double(uint64_t *heapAddress, double num)
+{
+#ifdef ANSWERS
+	*heapAddress = (uint64_t)num;
+#else
+	EM_ASM(writeI53ToU64Clamped($0, $1), heapAddress, num);
+#endif
+}
+
+void writeI53ToU64Signaling_double(uint64_t *heapAddress, double num)
+{
+#ifdef ANSWERS
+	*heapAddress = (uint64_t)num;
+#else
+	EM_ASM(try {
+		writeI53ToU64Signaling($0, $1)
+	} catch(e) {
+		HEAPU32[$0>>2] = 0x01020304;
+		HEAPU32[$0+4>>2] = 0x01020304;
+	} , heapAddress, num);
+#endif
+}
+
+int64_t readI53FromI64_toInt64(int64_t *heapAddress)
+{
+#ifdef ANSWERS
+	return *heapAddress;
+#else
+	return (int64_t)EM_ASM_DOUBLE(return readI53FromI64($0), heapAddress);
+#endif
+}
+
+double readI53FromI64(int64_t *heapAddress)
+{
+#ifdef ANSWERS
+	return (double)*heapAddress;
+#else
+	return EM_ASM_DOUBLE(return readI53FromI64($0), heapAddress);
+#endif
+}
+
+int64_t readI53FromU64_toInt64(uint64_t *heapAddress)
+{
+#ifdef ANSWERS
+	return (int64_t)*heapAddress;
+#else
+	return (int64_t)EM_ASM_DOUBLE(return readI53FromU64($0), heapAddress);
+#endif
+}
+
+double readI53FromU64(uint64_t *heapAddress)
+{
+#ifdef ANSWERS
+	return (double)*heapAddress;
+#else
+	return EM_ASM_DOUBLE(return readI53FromU64($0), heapAddress);
+#endif
+}
+
+int64_t convertI32PairToI53(int32_t lo, int32_t hi)
+{
+#ifdef ANSWERS
+	uint64_t val = (uint32_t)lo;
+	val |= ((uint64_t)(uint32_t)hi) << 32;
+	return (int64_t)val;
+#else
+	return (int64_t)EM_ASM_DOUBLE(return convertI32PairToI53($0, $1), lo, hi);
+#endif
+}
+
+int64_t convertU32PairToI53(uint32_t lo, uint32_t hi)
+{
+#ifdef ANSWERS
+	uint64_t val = (uint32_t)lo;
+	val |= ((uint64_t)(uint32_t)hi) << 32;
+	return val;
+#else
+	return (int64_t)EM_ASM_DOUBLE(return convertU32PairToI53($0, $1), lo, hi);
+#endif
+}
+
+int64_t testconvertI32PairToI53(int64_t val)
+{
+	uint64_t u = (uint64_t)val;
+	int32_t lo = (uint32_t)u;
+	int32_t hi = u >> 32;
+	return convertI32PairToI53(lo, hi);
+}
+
+int64_t testconvertU32PairToI53(uint64_t val)
+{
+	uint32_t lo = (uint32_t)val;
+	uint32_t hi = val >> 32;
+	return convertU32PairToI53(lo, hi);
+}
+
+int64_t testWriteI64AsI53(int64_t num)
+{
+	int64_t addr = 0;
+	writeI53ToI64_int64(&addr, num);
+	return addr;
+}
+
+int64_t testReadWriteI64AsI53(int64_t num)
+{
+	int64_t addr = 0;
+	writeI53ToI64_int64(&addr, num);
+	return readI53FromI64_toInt64(&addr);
+}
+
+uint64_t testReadWriteU64AsI53(uint64_t num)
+{
+	uint64_t addr = 0;
+	writeI53ToI64_int64((int64_t*)&addr, num);
+	return readI53FromU64_toInt64(&addr);
+}
+
+int main()
+{
+	// We can subdivide the set of all possible double precision floating point numbers + 64-bit (u)int numbers to eight categories:
```

Comment:
I know we are far from consistent but I'm trying to move us towards a consistent C++ style.

I added a .clang-format file in #7164 (based on the binaryen one) so you should be able to run `git clang-format origin/master` to reformat this code.  I know that most of out test code is not consistent with this style, so objecting on that basis seems reasonable but in the long run I'd like to converge.

Reply:
Looks like not in my git version yet :/
```
E:\code\emsdk\emscripten\master>git clang-format origin/master
git: 'clang-format' is not a git command. See 'git --help'.

E:\code\emsdk\emscripten\master>git --version
git version 2.24.1.windows.2
```

Reply:
clang-format is a separate tool that is part of llvm.  I think it works by installing itself as `git-clang-format.exe` on the PATH which `git` then picks up as sub-command.   The regular clang-format formats entire files, but `git-clang-format` only formats the lines you touch in your diff.

Its extremely useful so I would recommend trying to setting it up, but i won't block this PR on that.